File: TPraticas/Aula13/Input.java

import java.util.*;

public class Input {
    public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);

    int sz = getArraySize(scan);
    String[] names = getNames(scan,sz);

    int which = getWhich(scan);
    String aName = getName(which,names);

    System.out.println("Nome escolhido: "+aName);
  }

  public static int getArraySize(Scanner scan) {
    System.out.print("Quantos nomes? ");
    int n =  scan.nextInt();
    scan.nextLine();
    return n;
  }

  public static String[] getNames(Scanner scan, int sz) {
    String[] names = new String[sz];
    for (int i = 0; i < sz; i++ ){
      System.out.print("Insira nome #"+(i+1)+": ");
      names[i] = scan.nextLine();
    }
    return names;
  }

  public static int getWhich(Scanner scan) {
    System.out.print("Escolha o nome #: ");
    int x = scan.nextInt();
    return x;
  }

  public static String getName(int n,String[] vals) {
      return vals[n-1];
  }
}

File: TPraticas/Aula13/Input.cpp
/* 
 * File:   InputValidation2.cpp
 * Author: jepm
 *
 * Created on November 13, 2014, 11:32 AM
 */

#include <iostream>
using namespace std;

void getNames(string[], int);
int getWhich();
string getName(int, string[],int);
const int SIZE=5;

int main()
{
  int which;
  string names[SIZE];
  getNames(names,SIZE);

  which =getWhich();
  string aName = getName(which,names,SIZE);
  cout << "Escolheu o nome: "  << aName;

  return 0;
}

void getNames(string names[],int sz)
{
  for (size_t i = 0; i < sz; i++ )
  {
    cout << "Introduza nome #" << i+1 <<": ";
    cin >> names[i];
  }
}

int getWhich()
{
  size_t x;
  cout << "Escolha o nome #: ";
  cin >> x;
  return x;
}

string getName(int n,string vals[],int sz)
{
    if (n >=1 && n <= sz)
      return vals[n-1];
    else
      return "";
}
